### PR TITLE
gh/workflows: Use cilium-cli GHA to install CLI exec

### PR DESCRIPTION
**Pull request by brb**
This reduces not only the boilerplate, but allows with a few cmds to change the CLI vsn to test WIP changes. For example, to use a custom CLI CI build for all workflows:

        $ CI_BUILD_VSN=foobar
        $ git grep -l cilium_cli_version: | xargs sed -i 's/cilium_cli_version: .*/cilium_cli_version:/g'
        $ git grep -l cilium_cli_ci_version: | xargs sed -i 's/cilium_cli_ci_version:.*/cilium_cli_ci_version: $CI_BUILD_VSN/g'

Depends on https://github.com/cilium/cilium-cli/pull/1563

### CODE REVIEW

File: .github/workflows/conformance-aks-v1.11.yaml
Comment (aanm) [inline .github/workflows/conformance-aks-v1.11.yaml]: If I'm not mistaken, this will make dependanbot to update cilium/cilium-cli SHA on a weekly basis, or even revert to the last stable release.

**Review by aanm — CHANGES_REQUESTED** [review]
Adding Tobias as reviewer as well.

File: .github/workflows/conformance-aks-v1.11.yaml
Comment (tklauser) [reply .github/workflows/conformance-aks-v1.11.yaml]: Isn't the SHA only for the action? I think the version of cilium-cli binary being installed is still the one specified in `cilium_cli_version` (or `cilium_cli_ci_version`).

Comment (tklauser) [reply .github/workflows/conformance-aks-v1.11.yaml]: Ah, I guess this is what you meant :sweat_smile: the GH action is not yet part of a released version.

I plan to tag v0.14.1 later today, so we could use that SHA then.

Comment (brb) [reply .github/workflows/conformance-aks-v1.11.yaml]: To avoid frequent updates, can't we stick to the `main`? I know about security concerns, but in this case we are controlling the `cilium/cilium-cli` repo.

Comment (brb) [reply .github/workflows/conformance-aks-v1.11.yaml]: Changed to v0.14.1. Eagerly waiting for the release (a GHA cannot be used from unrealeased tag).

Comment (tklauser) [reply .github/workflows/conformance-aks-v1.11.yaml]: https://github.com/cilium/cilium-cli/releases/tag/v0.14.1

Comment (brb) [reply .github/workflows/conformance-aks-v1.11.yaml]: Thanks! Updated. PTAL.

File: .github/workflows/conformance-aks-v1.11.yaml
Comment (tklauser) [inline .github/workflows/conformance-aks-v1.11.yaml:75]: Do we really need to set this in every workflow? Can't we just keep `with.relase-version` when using the action below and rely on the `with.ci-version` default value being empty?

Comment (brb) [reply .github/workflows/conformance-aks-v1.11.yaml:75]: It's to make sed'ing easier (described in the description). I'd like to avoid setting it too, but didn't come up with any nice solution.

**Review by tklauser — APPROVED** [review]

**Review by nebril — APPROVED** [review]

File: .github/workflows/conformance-aks-v1.12.yaml
Comment (aanm) [inline .github/workflows/conformance-aks-v1.12.yaml]: I think this needs to be the SHA of the v0.14.1 tag so that renovate bot can update this dependency automatically.

**Review by aanm — APPROVED** [review]
Non blocking comment

File: .github/workflows/conformance-aks-v1.12.yaml
Comment (joestringer) [reply .github/workflows/conformance-aks-v1.12.yaml]: Is this nonblocking because we actually don't care and it's up to the opinion of the contributor, or it's nonblocking because we think it must be done and it just doesn't need to be done in this PR? I was under the impression that we exclusively use SHAs for supply chain security reasons, so we should just fix this up now before merge...?

**Comment by joestringer** [comment]
Removed ready-to-merge pending resolution of https://github.com/cilium/cilium/pull/25228#discussion_r1185159693 .

File: .github/workflows/conformance-aks-v1.12.yaml
Comment (brb) [reply .github/workflows/conformance-aks-v1.12.yaml]: My reasoning is that we control `cilium/cilium-cli`, so SHA is not needed. Also, if we use SHA, then wouldn't any merge to cilium-cli would cause the action update PR?

Comment (tklauser) [reply .github/workflows/conformance-aks-v1.12.yaml]: > Also, if we use SHA, then wouldn't any merge to cilium-cli would cause the action update PR?

Nope, only releases would cause renovatebot to bump the action. It would still use the SHA though. So I think there is no harm in using the release SHA.

Comment (brb) [reply .github/workflows/conformance-aks-v1.12.yaml]: Thanks. Updated.

**Comment by brb** [comment]
> Removed ready-to-merge pending resolution of https://github.com/cilium/cilium/pull/25228#discussion_r1185159693 .

@joestringer The comment has been addressed.

File: .github/workflows/conformance-aks-v1.12.yaml
Comment (tklauser) [reply .github/workflows/conformance-aks-v1.12.yaml]: > > Also, if we use SHA, then wouldn't any merge to cilium-cli would cause the action update PR?
> 
> Nope, only releases would cause renovatebot to bump the action. It would still use the SHA though. So I think there is no harm in using the release SHA.

Sorry, I forgot to mention that in order for renovate to only update digests on releases a comment with the version is needed, see https://docs.renovatebot.com/modules/manager/github-actions/

I've opened #25369 to add those.

**Pull request merged**
